packages/esm-patient-chart-app/src/clinical-views/utils/helpers.ts

```diff
@@ -0,0 +1,189 @@
+  if (isDate) return formatDate(encounter[param]);
+  else return encounter[param] ? encounter[param] : '--';
+}
+
+export function obsArrayDateComparator(left: Observation, right: Observation) {
+  return new Date(right.obsDatetime).getTime() - new Date(left.obsDatetime).getTime();
+}
+
+export function findObs(encounter: Encounter, obsConcept: string): Observation {
+  const allObs = encounter?.obs?.filter((observation) => observation.concept.uuid === obsConcept) || [];
+  return allObs?.length == 1 ? allObs[0] : allObs?.sort(obsArrayDateComparator)[0];
+}
+
+export function getObsFromEncounters(encounters: Encounter, obsConcept: string) {
+  const filteredEnc = encounters?.find((enc) => enc.obs.find((obs) => obs.concept.uuid === obsConcept));
+  return getObsFromEncounter(filteredEnc, obsConcept);
+}
+
+export function resolveValueUsingMappings(encounter: Encounter, concept: string, mappings) {
+  const obs = findObs(encounter, concept);
+  for (const key in mappings) {
+    if (typeof obs?.value === 'object' && 'uuid' in obs.value && mappings[key] === obs?.value?.uuid) {
+      return key;
+    }
+  }
+  return '--';
+}
+
+export function getConditionalConceptValue(encounter: Encounter, conditionalConceptMappings, isDate: Boolean) {
+  const { trueConcept, nonTrueConcept, dependantConcept, conditionalConcept } = conditionalConceptMappings;
+  const dependantValue = findObs(encounter, dependantConcept)?.value;
+  const dependantUuid =
+    typeof dependantValue === 'object' && 'uuid' in dependantValue ? dependantValue.uuid : undefined;
+  const finalConcept = dependantUuid === conditionalConcept ? trueConcept : nonTrueConcept;
+  return getObsFromEncounter(encounter, finalConcept, isDate);
+}
+
+export function getConceptFromMappings(encounter: Encounter, concepts: Array<string>) {
+  for (const concept of concepts) {
+    const obs = findObs(encounter, concept);
+    if (obs && obs.value) {
+      return concept;
+    }
+  }
+  return null;
+}
+
+export function getMultipleObsFromEncounter(encounter: Encounter, obsConcepts: Array<string>) {
+  let observations = [];
+  obsConcepts.map((concept) => {
+    const obs = getObsFromEncounter(encounter, concept);
+    if (obs !== '--') {
+      observations.push(obs);
+    }
+  });
+
+  return observations.length ? observations.join(', ') : '--';
+}
+
+export function getObsFromEncounter(
+  encounter: Encounter,
+  obsConcept: string,
+  isDate?: Boolean,
+  isTrueFalseConcept?: Boolean,
+  type?: string,
+  fallbackConcepts?: Array<string>,
+  secondaryConcept?: string,
+  t?: TFunction,
+) {
+  let obs = findObs(encounter, obsConcept);
+  if (!encounter || !obsConcept) {
+    return '--';
+  }
+
+  if (isTrueFalseConcept) {
+    if (typeof obs?.value === 'object') {
+      if (
+        (obs?.value?.uuid != esmPatientChartSchema.trueConceptUuid._default && obs?.value?.name?.name !== 'Unknown') ||
+        obs?.value?.name?.name === t('FALSE')
```

Comment:
This is a bad use of the translation system, since it requires that the translation for `FALSE` be whatever the name of the concept is in the locale. Just collect the UUID for the false concept via config.